spec/controllers/translator_controller_spec.rb

```diff
@@ -62,13 +79,25 @@
           expect(response).to have_http_status(:unprocessable_entity)
         end
 
-        it "returns the translated text" do
-          DiscourseTranslator::Microsoft.expects(:translate).with(reply).returns(%w[ja ニャン猫])
+        describe "all groups can translate" do
+          include_examples "translation_successful"
+        end
 
-          post :translate, params: { post_id: reply.id }, format: :json
+        describe "user is in a allowlisted group" do
+          before do
+            SiteSetting.restrict_translation_by_group =
+              "#{Group.find_by(name: "admins").id}|not_in_the_list"
+          end
+          let!(:user) { log_in :admin }
+          include_examples "translation_successful"
+        end
 
-          expect(response).to have_http_status(:ok)
-          expect(response.body).to eq({ translation: "ニャン猫", detected_lang: "ja" }.to_json)
+        describe "user is not in a allowlisted group" do
+          before { SiteSetting.restrict_translation_by_group = "not_in_the_list" }
+          it "should deny request to translate" do
+            response = post :translate, params: { post_id: 1 }, format: :json
+            expect(response).to have_http_status(:bad_request)
```

Comment:
what status code is this request returning? I believe it should be 401 (unauthorized) vs 400 (bad request).

Reply:
Good catch, not sure why I never looked into it